pkg/observability/opentelemetry_test.go

```diff
@@ -0,0 +1,239 @@
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package observability
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+
+	h "github.com/aws/aws-node-termination-handler/pkg/test"
+)
+
+var (
+	mockNth         = "aws.node.termination.handler"
+	mockErrorEvent  = "mockErrorEvent"
+	mockAction      = "cordon-and-drain"
+	mockNodeName1   = "nodeName1"
+	mockNodeName2   = "nodeName2"
+	mockNodeName3   = "nodeName3"
+	mockEventID1    = "eventID1"
+	mockEventID2    = "eventID2"
+	mockEventID3    = "eventID3"
+	successStatus   = "success"
+	errorStatus     = "error"
+	mockDefaultPort = 9092
+	mockClosedPort  = 9093
+)
+
+func TestInitMetrics(t *testing.T) {
+	getMetrics(t)
+
+	rr := mockMetricsRequest()
+
+	validateStatus(t, rr)
+
+	metricsMap := getMetricsMap(rr.Body.String())
+
+	runtimeMetrics := []string{
+		"go_gc_gogc_percent",
+		"go_memstats_frees_total",
+		"go_goroutines",
+	}
+
+	for _, metricName := range runtimeMetrics {
+		_, exists := metricsMap[metricName]
+		h.Assert(t, exists, fmt.Sprintf("%v metric should be present", metricName))
+	}
+}
+
+func TestErrorEventsInc(t *testing.T) {
+	metrics := getMetrics(t)
+
+	metrics.ErrorEventsInc(mockErrorEvent)
+
+	rr := mockMetricsRequest()
+
+	validateStatus(t, rr)
+
+	metricsMap := getMetricsMap(rr.Body.String())
+
+	validateEventErrorTotal(t, metricsMap, 1)
+	validateActionTotalV2(t, metricsMap, 0, successStatus)
+	validateActionTotalV2(t, metricsMap, 0, errorStatus)
+}
+
+func TestNodeActionsInc(t *testing.T) {
+	metrics := getMetrics(t)
+
+	metrics.NodeActionsInc(mockAction, mockNodeName1, mockEventID1, nil)
+	metrics.NodeActionsInc(mockAction, mockNodeName2, mockEventID2, nil)
+	metrics.NodeActionsInc(mockAction, mockNodeName3, mockEventID3, errors.New("mockError"))
+
+	rr := mockMetricsRequest()
+
+	validateStatus(t, rr)
+
+	metricsMap := getMetricsMap(rr.Body.String())
+
+	validateEventErrorTotal(t, metricsMap, 0)
+	validateActionTotalV2(t, metricsMap, 2, successStatus)
+	validateActionTotalV2(t, metricsMap, 1, errorStatus)
+}
+
+func TestRegisterMetricsWith(t *testing.T) {
+	const errorEventMetricsTotal = 23
+	const successActionMetricsTotal = 31
+	const errorActionMetricsTotal = 97
+
+	metrics := getMetrics(t)
+
+	errorEventLables := []attribute.KeyValue{labelEventErrorWhereKey.String(mockErrorEvent)}
```

Comment:
Typo: errorEventLabels. Same for other areas

Reply:
Fixed.